# Analyze placementdata - 10k.csv
Clean dataset to have numbers and normalize each column

In [2]:
import pandas as pd
from sklearn.preprocessing import MinMaxScaler
datasetFileName = "../data/placementdata - 10k.csv"

placementDataFrame = pd.read_csv(datasetFileName)

placementDataFrame.head()


StudentID  CGPA  Internships  Projects  Workshops/Certifications  \
0          1   7.5            1         1                         1   
1          2   8.9            0         3                         2   
2          3   7.3            1         2                         2   
3          4   7.5            1         1                         2   
4          5   8.3            1         2                         2   

   AptitudeTestScore  SoftSkillsRating ExtracurricularActivities  \
0                 65               4.4                        No   
1                 90               4.0                       Yes   
2                 82               4.8                       Yes   
3                 85               4.4                       Yes   
4                 86               4.5                       Yes   

  PlacementTraining  SSC_Marks  HSC_Marks PlacementStatus  
0                No         61         79       NotPlaced  
1               Yes         78         82          Placed  
2                No         79         80       NotPlaced  
3               Yes         81         80          Placed  
4               Yes         74         88          Placed

In [3]:
numericColumns = []
nonNumericColumns = []

for c in placementDataFrame.columns[1:]: #Exclude StudentID
    if(placementDataFrame[c].dtype == "object"):
        nonNumericColumns.append(c)
    else:
        numericColumns.append(c)

averages = {}
for c in numericColumns:
    l = len(placementDataFrame[c])
    s = 0
    for i in range(l):
        s += placementDataFrame[c][i]
    averages[c] = s/l

labelPercentages = {}
for c in nonNumericColumns:
    l = len(placementDataFrame[c])
    labels = {}
    for i in range(l):
        val = placementDataFrame[c][i]
        if val not in labels:
            labels[val] = 1
        else:
            labels[val] += 1
    
    s = 0
    for label in labels:
        s += labels[label]
    for label in labels:
        labels[label] = (labels[label]/s)*100

    labelPercentages[c] = labels


print("Averages:")
for c in averages:
    print(f"{c} - {averages[c]}")
print()
print("Percentages")


for c in labelPercentages:
    labels = labelPercentages[c]
    print(c)
    for label in labels:
        print(f"{label} - {labels[label]}", end="  ")
    print()

Averages:
CGPA - 7.69800999999997
Internships - 1.0492
Projects - 2.0266
Workshops/Certifications - 1.0132
AptitudeTestScore - 79.4499
SoftSkillsRating - 4.3239600000000395
SSC_Marks - 69.1594
HSC_Marks - 74.5015

Percentages
ExtracurricularActivities
No - 41.46  Yes - 58.540000000000006  
PlacementTraining
No - 26.82  Yes - 73.18  
PlacementStatus
NotPlaced - 58.03  Placed - 41.97  


# Data Analysis
## Averages:
- CGPA - 7.69800999999997
- Internships - 1.0492
- Projects - 2.0266
- Workshops/Certifications - 1.0132
- AptitudeTestScore - 79.4499
- SoftSkillsRating - 4.3239600000000395
- SSC_Marks - 69.1594
- HSC_Marks - 74.5015

## Percentages
### ExtracurricularActivities
 - No - 41.46  Yes - 58.540000000000006  
### PlacementTraining
 - No - 26.82  Yes - 73.18  
### PlacementStatus
 - NotPlaced - 58.03  Placed - 41.97 

In [4]:
# Mapping text to numbers

placementDataFrame["ExtracurricularActivities"] = (
    placementDataFrame["ExtracurricularActivities"]
    .str.strip()         
    .str.lower()         
    .map({"yes": 1, "no": 0})
)

placementDataFrame["PlacementTraining"] = (
    placementDataFrame["PlacementTraining"]
    .str.strip()         
    .str.lower()         
    .map({"yes": 1, "no": 0})
)

placementDataFrame["PlacementStatus"] = (
    placementDataFrame["PlacementStatus"]
    .str.strip()         
    .str.lower()         
    .map({"notplaced": 0 , "placed": 1})
)

# Normalizing all columns to be between 0 and 1
scaler = MinMaxScaler()
numeric_cols = ['CGPA', 'Internships', 'Projects', 'Workshops/Certifications',
                'AptitudeTestScore', 'SoftSkillsRating', 'SSC_Marks', 'HSC_Marks']

placementDataFrame[numeric_cols] = scaler.fit_transform(placementDataFrame[numeric_cols])


In [5]:
# Assuming df is your DataFrame

# Drop StudentID (not a predictive feature)
X = placementDataFrame.drop(columns=["StudentID", "PlacementStatus", "AptitudeTestScore", "Workshops/Certifications", ])
y = placementDataFrame["PlacementStatus"]

print(X.head())

from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Initialize and train model
model = LogisticRegression()
model.fit(X_train, y_train)

# Predict and evaluate
y_pred = model.predict(X_test)

print("Accuracy:", accuracy_score(y_test, y_pred))
print(classification_report(y_test, y_pred))


       CGPA  Internships  Projects  SoftSkillsRating  \
0  0.384615          0.5  0.333333          0.777778   
1  0.923077          0.0  1.000000          0.555556   
2  0.307692          0.5  0.666667          1.000000   
3  0.384615          0.5  0.333333          0.777778   
4  0.692308          0.5  0.666667          0.833333   

   ExtracurricularActivities  PlacementTraining  SSC_Marks  HSC_Marks  
0                          0                  0   0.171429   0.709677  
1                          1                  1   0.657143   0.806452  
2                          1                  0   0.685714   0.741935  
3                          1                  1   0.742857   0.741935  
4                          1                  1   0.542857   1.000000  
Accuracy: 0.785
              precision    recall  f1-score   support

           0       0.83      0.80      0.81      1172
           1       0.73      0.76      0.75       828

    accuracy                           0.79      20

## Different Models

### Logistic Regression Model
- 79% accuracy with Logistic Regression Model excluding features which are dropped

# Analyse student career suggestion - 20k.csv

In [3]:
import pandas as pd

suggestionDataFileName = "../data/student career suggestion - 20k.csv"

suggestionDF = pd.read_csv(suggestionDataFileName)

suggestionDF.head()


Acedamic percentage in Operating Systems  percentage in Algorithms  \
0                                        69                        63   
1                                        78                        62   
2                                        71                        86   
3                                        76                        87   
4                                        92                        62   

   Percentage in Programming Concepts  Percentage in Software Engineering  \
0                                  78                                  87   
1                                  73                                  60   
2                                  91                                  87   
3                                  60                                  84   
4                                  90                                  67   

   Percentage in Computer Networks  Percentage in Electronics Subjects  \
0                               94                                  94   
1                               71                                  70   
2                               61                                  81   
3                               89                                  73   
4                               71                                  89   

   Percentage in Computer Architecture  Percentage in Mathematics  \
0                                   87                         84   
1                                   73                         84   
2                                   72                         72   
3                                   62                         88   
4                                   73                         71   

   Percentage in Communication skills  Hours working per day  ...  \
0                                  61                      9  ...   
1                                  91                     12  ...   
2                                  94                     11  ...   
3                                  69                      7  ...   
4                                  73                      4  ...   

   Interested Type of Books  Salary Range Expected  In a Realtionship?  \
0              Prayer books                 salary                  no   
1                 Childrens                 salary                 yes   
2                    Travel                   Work                  no   
3                   Romance                   Work                 yes   
4                 Cookbooks                 salary                  no   

   Gentle or Tuff behaviour? Management or Technical Salary/work  \
0                   stubborn              Management      salary   
1                     gentle               Technical      salary   
2                   stubborn              Management        work   
3                     gentle              Management        work   
4                   stubborn              Management        work   

  hard/smart worker worked in teams ever? Introvert  \
0       hard worker                   yes        no   
1       hard worker                    no       yes   
2       hard worker                    no       yes   
3      smart worker                   yes       yes   
4       hard worker                   yes       yes   

               Suggested Job Role  
0              Database Developer  
1            Portal Administrator  
2            Portal Administrator  
3  Systems Security Administrator  
4        Business Systems Analyst  

[5 rows x 39 columns]